dev/buildtool/spin_commands.py

```diff
@@ -312,39 +294,23 @@ def promote_spin(self, repository):
     git_dir = repository.git_dir
     git = self.__scm.git
 
-    match = re.match(r'(\d+)\.(\d+)\.(\d+)-\d+', self.__gate_version)
-    gate_major = match.group(1)
-    gate_min = match.group(2)
-    if match is None:
-      raise_and_log_error(
-          ConfigError('gate version {version} is not X.Y.Z-<buildnum>'
-                      .format(version=self.__gate_version)))
-
     if len(self.__spinnaker_version.split('.')) != 3:
       raise_and_log_error(
           ConfigError('Expected spinnaker version in the form X.Y.Z-N'))
 
-    # Note: spin CLI is coupled to the Gate major and minor version.
-    # Gate is a routing server, so features and breaking changes in Gate
-    # must be reflected in spin since it is a client. We pin only the major
-    # and minor versions so fixes (thus patch version) are decoupled between
-    # the two.
-    patch = '0' # Patch is reset on a new Gate major or minor.
-    tag_matcher = re.compile(r'version-{maj}.{min}.(\d+)'
-                             .format(maj=gate_major, min=gate_min))
-    tags = git.fetch_tags(git_dir)
-    tag_matches = [tag_matcher.match(t) for t in tags if tag_matcher.match(t)]
-    if tag_matches:
-      patch_versions = [int(m.group(1)) for m in tag_matches]
-      max_patch = max(patch_versions)
-      last_tag = 'version-{maj}.{min}.{max_patch}'.format(maj=gate_major,
-                                                          min=gate_min,
-                                                          max_patch=max_patch)
-      self.__no_changes = git.query_local_repository_commit_id(git_dir) == git.query_commit_at_tag(git_dir, last_tag)
-      patch = str(max_patch + 1)
-
-    self.__stable_version = '{major}.{minor}.{patch}'.format(
-        major=match.group(1), minor=match.group(2), patch=patch)
+    next_spin_semver = bump_spin_patch(git, git_dir, self.__gate_version)
```

Comment:
I think the logging messages in bump_spin_patch might get confusing in this context because they suggest you are creating new versions, but really you are not. You are just trying to identify what the 'last' version was to see if you dont need to do any work because there were no changes.

Reply:
Changed up the wording here to specify that we are calculating a 'next spin patch version'.